cpp/src/Ice/ObjectAdapterI.cpp

```diff
@@ -1238,49 +1219,65 @@ ObjectAdapterI::computePublishedEndpoints()
     vector<EndpointIPtr> endpoints;
     if (_routerInfo)
     {
-        //
         // Get the router's server proxy endpoints and use them as the published endpoints.
-        //
-        vector<EndpointIPtr> endps = _routerInfo->getServerEndpoints();
-        for (vector<EndpointIPtr>::const_iterator p = endps.begin(); p != endps.end(); ++p)
-        {
-            if (::find(endpoints.begin(), endpoints.end(), *p) == endpoints.end())
-            {
-                endpoints.push_back(*p);
-            }
-        }
+        endpoints = _routerInfo->getServerEndpoints();
     }
     else
     {
-        //
-        // Parse published endpoints. If set, these are used in proxies
-        // instead of the connection factory endpoints.
-        //
-        string endpts = _communicator->getProperties()->getProperty(_name + ".PublishedEndpoints");
-        endpoints = parseEndpoints(endpts, false);
+        // Parse published endpoints. If set, these are used in proxies instead of the connection factory endpoints.
+        endpoints = parseEndpoints(_communicator->getProperties()->getProperty(_name + ".PublishedEndpoints"), false);
         if (endpoints.empty())
         {
-            //
-            // If the PublishedEndpoints property isn't set, we compute the published endpoints
-            // from the OA endpoints, expanding any endpoints that may be listening on INADDR_ANY
-            // to include actual addresses in the published endpoints.
-            //
-            for (unsigned int i = 0; i < _incomingConnectionFactories.size(); ++i)
+            // If the PublishedEndpoints property isn't set, we compute the published endpoints from the factory
+            // endpoints.
+            for (const auto& factory : _incomingConnectionFactories)
+            {
+                endpoints.push_back(factory->endpoint());
+            }
+
+            // Remove all loopback endpoints
+            vector<EndpointIPtr> endpointsNoLoopback;
+            copy_if(
+                endpoints.begin(),
+                endpoints.end(),
+                back_inserter(endpointsNoLoopback),
+                [](const EndpointIPtr& endpoint) { return !endpoint->isLoopback(); });
+
+            // Retrieve published host
+            string publishedHost = _communicator->getProperties()->getProperty(_name + ".PublishedHost");
+
+            if (!endpointsNoLoopback.empty())
             {
-                vector<EndpointIPtr> endps = _incomingConnectionFactories[i]->endpoint()->expandIfWildcard();
-                for (vector<EndpointIPtr>::const_iterator p = endps.begin(); p != endps.end(); ++p)
+                endpoints = endpointsNoLoopback;
```

Comment:
Can we move assign here?
```suggestion
                endpoints = std::move(endpointsNoLoopback);
```

Reply:
Fixed as suggested.